src/hermit.rs

```diff
@@ -43,11 +43,38 @@ impl<T: Config> Hermit<T> {
         let path = self.config.shell_root_path().join(name);
         file_ops.create_git_repo(path);
     }
+
+    pub fn inhabit_shell(&mut self, file_ops: &mut FileOperations, name: &str) -> Result<Vec<String>, Error> {
+        let mut shell_names = Vec::new();
+        let mut res = Ok(());
+        if name == "" {
+            shell_names = self.get_shell_list();
+        } else {
+            res = self.set_current_shell(name);
+        }
+        match res {
+            Ok(v) => Ok(shell_names),
+            Err(e) => Err(Error::ShellDoesNotExist),
+        }
+    }
+
+    pub fn get_shell_list(&mut self) -> Vec<String> {
+        let mut shell_names = Vec::new();
+        let paths = fs::read_dir(self.config.shell_root_path().to_owned()).unwrap();
+        for path in paths {       
+            let shell_path = path.unwrap().path();
+            let shell_name = shell_path.file_name().unwrap().to_str().unwrap().to_owned();
+            shell_names.push(shell_name);
+            shell_names.sort();
```

Comment:
I am running into an error when I do that. Have been unable to resolve so far. This is the error I am getting
```
ermit.rs:70:16: 70:34 error: mismatched types:
 expected `collections::vec::Vec<collections::string::String>`,
    found `()`
(expected struct `collections::vec::Vec`,
    found ()) [E0308]
hermit.rs:70         return shell_names.sort();
                            ^~~~~~~~~~~~~~~~~~
hermit.rs:70:16: 70:34 help: run `rustc --explain E0308` to see a detailed explanation
```
any ideas?